interceptors/CORS.cfc

```diff
@@ -37,13 +37,7 @@ component {
             return;
         }
 
-        var allowedOrigins = settings.allowOrigins;
-        if ( isClosure( allowedOrigins ) || isCustomFunction( allowedOrigins ) ) {
-            allowedOrigins = allowedOrigins( event );
-        }
-        if ( ! isSimpleValue( allowedOrigins ) ) {
-            allowedOrigins = arrayToList( allowedOrigins, ", " );
-        }
+        var allowedOrigin = event.getHTTPHeader( "Origin", "" );
```

Comment:
I'm not sure this is actually what we want.  This will allow the origin they pass in to be allowed no matter what.  I think we need to check that the host is in the allowed list and return it then.

Also, we need to keep the closure/custom function logic because that is a supported path.

Reply:
Thanks for the review Eric but I think my suggested update is solid.

The code block I removed is already provided by the `isAllowed()` method called earlier by the `if` statement therefore the closure/custom function logic support is retained. My simple change is around what is subsequently passed along in the `preProcess()` event.

You'll see that I purposefully used the variable name `allowedOrigin` (singular) to capture the requested origin from the header rather than use the `allowedOrigins` (plural) which is the list of _allowable_ origins from the settings. I then used that in my other logging, cache and header changes to the code.

Admittedly I forgot to include a failure expectation in my tests. I have done that now and have pushed that change for review. For me, all tests pass.

Let me know what you think.

Reply:
Ah, I see now.  Thanks!